### PR TITLE
Fix/523 non disclosed salary filtering

### PR DESCRIPTION
**Pull request by SahilLamba0008**
This PR addresses a bug #523 in the job sorting functionality where sorting by `maxSalary` was not correctly pushing non-disclosed salary jobs to the end. The fix ensures that `nulls: 'last'` is applied only when sorting by `maxSalary`, while other fields (e.g., `postedAt`) use standard sorting logic without `nulls`. This resolves errors encountered when sorting by non-salary fields.

Fixes issue #523.

https://github.com/user-attachments/assets/91bf7dff-228a-458d-bc4f-acb0d893045c

### CODE REVIEW

**Comment by SahilLamba0008** [comment]
@VineeTagarwaL-code  review please.

File: next.config.js [hunk 1]
```diff
@@ -35,6 +35,10 @@ const nextConfig = {
         protocol: 'https',
         hostname: 'www.example.com',
       },
+      {
+        protocol: 'https',
+        hostname: 'images.unsplash.com'
+      }
```
Comment (VineeTagarwaL-code) [inline next.config.js:41]: why ?

**Review by VineeTagarwaL-code — CHANGES_REQUESTED** [review]

File: next.config.js (hunk 1 above)
Comment (SahilLamba0008) [reply next.config.js:41]: did it for some testing locally, will raise a new pr after removing it.

Comment (SahilLamba0008) [reply next.config.js:37]: @VineeTagarwaL-code please check now.

**Pull request merged**